prepare/metrics/llm_as_judge/pairwise_rating/llama_3_ibm_genai_arena_hard_template.py

```diff
@@ -10,27 +12,49 @@
     "templates.response_assessment.pairwise_comparative_rating.arena_hard",
     "templates.response_assessment.pairwise_comparative_rating.arena_hard_with_shuffling",
 ]
+
+inference_engines = [
+    ("ibm_wml", WMLInferenceEngine),
+    ("ibm_genai", IbmGenAiInferenceEngine),
+    ("generic_engine", GenericInferenceEngine),
+]
+
+
 for template in templates:
     task = "pairwise_comparative_rating.single_turn"
 
     for model_id in model_list:
-        inference_model = IbmGenAiInferenceEngine(
-            model_name=model_id, max_new_tokens=2048, random_seed=42
-        )
-        model_label = model_id.split("/")[1].replace("-", "_").replace(".", ",").lower()
-        model_label = f"{model_label}_ibm_genai"
-        template_label = template.split(".")[-1]
-        metric_label = f"{model_label}_template_{template_label}"
-        metric = LLMAsJudge(
-            inference_model=inference_model,
-            template=template,
-            task=task,
-            format=format,
-            main_score=metric_label,
-        )
-
-        add_to_catalog(
-            metric,
-            f"metrics.llm_as_judge.pairwise_comparative_rating.{model_label}_template_{template_label}",
-            overwrite=True,
-        )
+        for inference_engine_name, inference_engine in inference_engines:
+            if (
+                inference_engine_name == "ibm_wml"
+                and model_id == "meta-llama/llama-3-8b-instruct"
+            ):
+                continue  # currently not supported
+
+            # if inference engine is generic, these configurations will be defined when it is saved to the catalog
+            if inference_engine_name != "generic_engine":
+                inference_model = inference_engine(
+                    model_name=model_id, max_new_tokens=2048, random_seed=42
+                )
+            else:
+                inference_model = inference_engine()
```

Comment:
doesn't it need a default?

Reply:
I agree with Ofir. I think that we should add a default (I think that our main use is IBM genai), which will enable us to use mainly the generic metric. The internal users will use it the same way they used the metric today, and the external users will change the actual engine with the env var

Reply:
added